dev-tools/mage/gotest.go

```diff
@@ -93,6 +95,32 @@ func makeGoTestArgsForModule(name, module string) GoTestArgs {
 	return params
 }
 
+// fetchGoPackages uses "go list ./module/..." to get all Go packages for a
+// module.
+// It returns a list of packages that are relative to the module path.
+// Example: for module "kafka" it'll return:
+//
+//	[kafka kafka/broker kafka/consumer kafka/consumergroup kafka/partition kafka/producer]
+func fetchGoPackages(module string) ([]string, error) {
+	cmd := exec.Command("go", "list", "-tags", "integration", fmt.Sprintf("./%s/...", module)) //nolint:gosec // used only to run tests
```

Comment:
We should probably be using [x/sys/execabs](https://pkg.go.dev/golang.org/x/sys/execabs) instead of os/exec.